addons/account/migrations/9.0.1.1/pre-migration.py

```diff
@@ -225,7 +225,7 @@ def remove_account_moves_from_special_periods(cr):
         INSERT INTO account_move_line (company_id, account_id, credit, debit, date, journal_id, move_id, name, period_id, reconcile_id, reconcile_ref)
         SELECT aml.company_id, aml.account_id, CASE WHEN SUM(aml.debit - aml.credit)<0 THEN 0 ELSE SUM(aml.debit - aml.credit) END as debit, CASE WHEN SUM(aml.credit - aml.debit)<0 THEN 0 ELSE SUM(aml.credit - aml.debit) END as credit, aml.date, aml.journal_id, aml.move_id, 'Year opening entry' as name, aml.period_id, aml.reconcile_id, aml.reconcile_ref
         FROM account_move_line aml
-        WHERE id IN %s
+        WHERE id IN (%s)
```

Comment:
@StefanRijnhart I think the missing parenthesis was the reason of the issue. I'm investigating further.

Reply:
No, that is not the case
```
>>> self.env.cr.execute(
...     "SELECT id FROM res_users WHERE partner_id IN (%s)", (tuple(partners.ids),))
2021-05-21 09:27:15,151 27938 INFO migtestou9_20210520_2018 openerp.sql_db: Programming error: operator does not exist: integer = record
LINE 1: SELECT id FROM res_users WHERE partner_id IN ((3, 583692, 64...
                                                  ^
HINT:  No operator matches the given name and argument types. You might need to add explicit type casts.
, in query SELECT id FROM res_users WHERE partner_id IN (%s)
Traceback (most recent call last):
  File "<console>", line 2, in <module>
  File "/home/openupgrade/ou9/parts/odoo/openerp/sql_db.py", line 154, in wrapper
    return f(self, *args, **kwargs)
  File "/home/openupgrade/ou9/parts/odoo/openerp/sql_db.py", line 234, in execute
    res = self._obj.execute(query, params)
UndefinedFunction: operator does not exist: integer = record
LINE 1: SELECT id FROM res_users WHERE partner_id IN ((3, 583692, 64...
                                                  ^
HINT:  No operator matches the given name and argument types. You might need to add explicit type casts.
```

Reply:
Sorry, my bad!